### PR TITLE
HABI-25 JPA Repo + Data Loader Updates and Service Implementation for Goals

### PR DESCRIPTION
**Pull request by Schachte**
## Ticket Reference ([View Current Issues Here](https://github.com/Habicus/Habicus-Core/issues))
My Issue Number Is: #25 

##### Core Reviewers:
@Habicus/core-team

##### Merge Owner:
@mweser
@Habicus/core-team
@austinsorrells
@schachte

## Status
**Ready**

## Description
- Added new API layer for goal controller
- Added new config for bean injection
- Updated models to support nested elements using foreign keys in hibernate
- Ability to custom query goals using DAO

tl;dr We can hit an endpoint and return goals associated with a user

## Steps to Test or Reproduce
1. Run server
2. Visit localhost:7777/api/v1/goal/{:id}
3. Verify results are present

### CODE REVIEW

File: src/main/java/com/habicus/core/service/Goal/GoalService.java [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
 public class GoalService {
 
+  @Autowired private GoalRepository goalRepository;
+
+  public List<Goal> retrieveGoalsByUserId(Long id) {
+    List<Goal> goals = goalRepository.findGoalByUserId(id);
```
Comment (austinsorrells) [inline src/main/java/com/habicus/core/service/Goal/GoalService.java]: should goalRepository.findGoalByUserId() instead be named findGoalsByUserId()?

File: src/main/java/com/habicus/core/dao/repository/GoalRepository.java [hunk 2]
```diff
@@ -18,8 +18,11 @@
...
 
 import com.habicus.core.model.Goal;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface GoalRepository extends JpaRepository<Goal, Long> {}
+public interface GoalRepository extends JpaRepository<Goal, Long> {
+  List<Goal> findGoalByUserId(Long id);
```
Comment (austinsorrells) [inline src/main/java/com/habicus/core/dao/repository/GoalRepository.java]: Should this be findGoalsByUserId()?

**Comment by Schachte** [comment]
@austinsorrells Updated
